ballerina/caller.bal

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+
+# Represents an FTP caller that is passed to the onFileChange function
+public isolated client class Caller {
+    private final Client 'client;
+
+    # Gets invoked during object initialization.
+    #
+    # + 'client - The `ftp:Client` which is used to interact with the Ftp server
+    # + return - `ftp:Error` in case of errors or `()` otherwise
```

Comment:
We don't return any errors, right?

Reply:
Updated in https://github.com/ballerina-platform/module-ballerina-ftp/pull/580/commits/af693601b9db53582123f279e485dd8fed286a4e